src/qe/qe_mbp.cpp

```diff
@@ -405,8 +406,18 @@ class mbproj::impl {
         return true;
     }
 
+    bool has_unsupported_th(const expr_ref_vector fmls) {
```

Comment:
shouldn't you use allow-lists of supported theories instead of blocked lists of unsupported theories? Someone will then start fuzzing this using the special relations theory or user-propagator operators?